### DISCUSSION TITLE
Problems with group.set.volume

### DISCUSSION THREAD

**Discussion by CH-1967** [hklages/node-red-contrib-sonos-plus#273] [Q&A] [answered]
I have noticed when I use group.set.volume,
that not all speakers in the group have the same volume.
If I set the value to 5, 3 speakers have the value 6. 
I consider value 11 as ok.
But 3 speakers have the following values 3, 3 and 7.
If I set the value to 15, they will have 11, 11 and 19.
These three differ a bit more from the set value.

With player.set.volume on each speaker the values are set correctly.

The type of the speakers are all the same. It can't be because of that.

Here is my test.

`[
    {
        "id": "4802e8fed372b428",
        "type": "sonos-universal",
        "z": "303cbe1e3fa8e2c6",
        "confignode": "1e6c16629b286555",
        "command": "group.set.volume",
        "state": "",
        "stateType": "str",
        "avoidCheckPlayerAvailability": false,
        "name": "",
        "x": 1290,
        "y": 460,
        "wires": [
            [
                "9082c985cbd9c972"
            ]
        ]
    },
    {
        "id": "9ea260e623836c27",
        "type": "inject",
        "z": "303cbe1e3fa8e2c6",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "playerName",
                "v": "WZ",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "15",
        "payloadType": "num",
        "x": 1090,
        "y": 460,
        "wires": [
            [
                "4802e8fed372b428"
            ]
        ]
    },
    {
        "id": "9082c985cbd9c972",
        "type": "debug",
        "z": "303cbe1e3fa8e2c6",
        "name": "debug 91",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "true",
        "targetType": "full",
        "statusVal": "",
        "statusType": "auto",
        "x": 1540,
        "y": 460,
        "wires": []
    },
    {
        "id": "808162322def3dcb",
        "type": "sonos-universal",
        "z": "303cbe1e3fa8e2c6",
        "confignode": "1e6c16629b286555",
        "command": "household.create.group",
        "state": "",
        "stateType": "num",
        "avoidCheckPlayerAvailability": false,
        "name": "",
        "x": 1310,
        "y": 400,
        "wires": [
            [
                "091ca0b5973a752d"
            ]
        ]
    },
    {
        "id": "e13c8a632ddd77b8",
        "type": "inject",
        "z": "303cbe1e3fa8e2c6",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "WZ,Buero1,Buero2,Bad,SZ,GZ",
        "payloadType": "str",
        "x": 1090,
        "y": 400,
        "wires": [
            [
                "808162322def3dcb"
            ]
        ]
    },
    {
        "id": "091ca0b5973a752d",
        "type": "debug",
        "z": "303cbe1e3fa8e2c6",
        "name": "debug 93",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "true",
        "targetType": "full",
        "statusVal": "",
        "statusType": "auto",
        "x": 1540,
        "y": 400,
        "wires": []
    },
    {
        "id": "1e6c16629b286555",
        "type": "sonos-config",
        "name": "WZ",
        "serialnum": "",
        "ipaddress": "192.168.75.52"
    }
]`

**Comment by hklages** [comment, marked as answer]
Right. That's a bit confusing but works as designed (by SONOS).  [See this](https://github.com/hklages/node-red-contrib-sonos-plus/wiki#group-volume).

You can set the relations by using the command [group.create.volumesnap](https://github.com/hklages/node-red-contrib-sonos-plus/wiki/A.1-Universal-Node#groupcreatevolumesnap)

**Comment by CH-1967** [comment]
Thank you. That explains my problem.
Tested it with group.create.volumesnap. 
Everything looks good.
Will test it extensively in the next few days.
